In [27]:
from transformers import T5Tokenizer, T5ForConditionalGeneration
from safetensors.torch import load_file
import torch

model_name = "t5-small"  
tokenizer = T5Tokenizer.from_pretrained(model_name)
model = T5ForConditionalGeneration.from_pretrained(model_name)

adapter_weights = load_file("results/checkpoint-2250/model.safetensors")

model.load_state_dict(adapter_weights, strict=False)



target_goodness_score = 20  
prompt = f"Generate one a new 12-quarter schedule maximizing goodness score. Target cumulative goodness score: {target_goodness_score}."


inputs = tokenizer(prompt, return_tensors="pt")

# Make the output to be more funky
torch.manual_seed(torch.randint(0, 10000, (1,)).item())

output_sequences = model.generate(
    input_ids=inputs["input_ids"],
    max_length=512,          
    num_beams=1,              
    temperature=0.8,          
    do_sample=True,           
    top_k=100,                
    top_p=0.95                
)


generated_schedule = tokenizer.decode(output_sequences[0], skip_special_tokens=True)
print("Generated Schedule:", generated_schedule)



Generated Schedule: Quarter 1: ['MATH 19A', 'CSE 20']. Score: 2.5. Quarter 2: ['MATH 19B', 'CSE 12']. Score: 0.4000000000000003. Quarter 3: ['AM 30', 'CSE 107']. Score: -1.5. Quarter 4: ['CSE 12', 'CSE 16']. Score: 0.0. Quarter 5: ['CSE 13S', 'ECE 30']. Score: 0.0. Quarter 6: ['CSE 40', 'CSE 30']. Score: 0.0. Quarter 7: ['CSE 101', 'CSE 185S']. Score: 0.0. Quarter 9: ['CSE 112', 'CSE 130']. Score: 0.0. Quarter 9: ['CSE 101M', 'CSE 101M']. Score: 0.0. Quarter 10: ['ELECTIVE', 'CSE 130']. Score: 0.0. Quarter 11: ['ELECTIVE']. Score: 0.0. Quarter 12: ['ELECTIVE', 'CAPSTONE']. Score: 0.0.


Generated Schedule: Quarter 1: ['MATH 19A', 'CSE 20']. Score: 2.5. Quarter 2: ['MATH 19B', 'CSE 16']. Score: 0.7999999999999998. Quarter 3: ['AM 30', 'CSE 12']. Score: 0.7999999999999998. Quarter 4: ['CSE 30', 'CSE 30']. Score: 0.0. Quarter 5: ['CSE 13S', 'CSE 13S']. Score: 0.0. Quarter 6: ['CSE 101', 'CSE 120']. Score: 0.0. Quarter 7: ['CSE 130', 'CSE 101M']. Score: 0.0. Quarter 8: ['CSE 130', 'CSE 130']. Score: 0.0. Quarter 9: ['CSE 101M', 'CSE 130']. Score: 0.0. Quarter 10: ['ELECTIVE', 'CSE 130']. Score: 0.0. Quarter 11: ['ELECTIVE']. Score: 0.0. Quarter 12: ['ELECTIVE', 'CAPSTONE']. Score: 0.0.


Generated Schedule: Quarter 1: ['MATH 20A', 'CSE 20']. Score: 2.5. Quarter 2: ['MATH 20B', 'CSE 16']. Score: 0.7999999999999998. Quarter 3: ['MATH 23A', 'CSE 16']. Score: 0.7999999999999998. Quarter 4: ['CSE 16', 'CSE 13S']. Score: 0.0. Quarter 5: ['CSE 12', 'CSE 12']. Score: 0.0. Quarter 6: ['CSE 13S']. Score: 0.0. Quarter 8: ['CSE 101', 'CSE 130']. Score: 0.0. Quarter 8: ['CSE 130', 'CSE 103']. Score: 0.0. Quarter 9: ['CSE 112', 'CSE 101M']. Score: 0.0. Quarter 10: ['ELECTIVE', 'CSE 185S']. Score: 0.0. Quarter 11: ['ELECTIVE']. Score: 0.0. Quarter 12: ['ELECTIVE', 'CAPSTONE']. Score: 0.0.


Generated Schedule: Quarter 1: ['MATH 19A', 'CSE 20']. Score: 2.5. Quarter 2: ['MATH 19B', 'CSE 12']. Score: 0.4000000000000003. Quarter 3: ['AM 30', 'CSE 107']. Score: -1.5. Quarter 4: ['CSE 12', 'CSE 16']. Score: 0.0. Quarter 5: ['CSE 13S', 'ECE 30']. Score: 0.0. Quarter 6: ['CSE 40', 'CSE 30']. Score: 0.0. Quarter 7: ['CSE 101', 'CSE 185S']. Score: 0.0. Quarter 9: ['CSE 112', 'CSE 130']. Score: 0.0. Quarter 9: ['CSE 101M', 'CSE 101M']. Score: 0.0. Quarter 10: ['ELECTIVE', 'CSE 130']. Score: 0.0. Quarter 11: ['ELECTIVE']. Score: 0.0. Quarter 12: ['ELECTIVE', 'CAPSTONE']. Score: 0.0.
